flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/CommonTestUtils.java

```diff
@@ -198,16 +205,17 @@ public static void assertThrows(
     @SuppressWarnings("BusyWait")
     public static void waitUtil(Supplier<Boolean> condition, Duration timeout, String errorMsg)
             throws TimeoutException, InterruptedException {
-        long timeoutMs = timeout.toMillis();
-        if (timeoutMs <= 0) {
+        long timeoutNanos = timeout.toNanos();
+        if (timeoutNanos <= 0) {
             throw new IllegalArgumentException("The timeout must be positive.");
         }
-        long startingTime = System.currentTimeMillis();
-        while (!condition.get() && System.currentTimeMillis() - startingTime < timeoutMs) {
+        long startingTime = System.nanoTime();
```

Comment:
Can we change this to use `Deadline` as well, to set a good example to not manually use the time functions (which is safer, many contributors get the time functions wrong accidentally).

Reply:
Did you means wrong `System.currentTimeMillis` ? I think it is implementor's responsibility. As a user, I think `Duration` in api is more friendly, there are many wrapping method `Deadline.fromNow(Duration.of(...))`.

I guess I could use `Deadline` in implementation first ?

Reply:
Hmmm, this `CommonTestUtils` resides in `flink-test-utils-junit` package which does not depend on `flink-core` where `Deadline` lives in. More worse, `flink-core` depends on `-junit`. I guess we encounter this due to fact that `flink-test-utils` is more flink related(with scala dependencies) while `-junit` has not flink dependencies. 

I guess we have to stop here for this `Duration` version for now.

I think there are candidates for follow up:
* Move `CommonTestUtils`(and other no junit related ?) to `flink-test-utils`. Dependants of `flink-test-utils-junit` are more than `flink-test-utils`. No sure whether some of them might want flink agnostic on purpose.
* Simple renaming to avoid naming clashing on usages.
* Others basing on more through investigation.

Reply:
Or I could delete `CommonTestUtils.waitUntil` from `flink-test-utils-junit` for now ? I tried, all dependants also depend on `flink-test-utils` currently.

Reply:
I think we have options here:
1. Keep it as is.
2. Migrate `CommonTestUtils.waitUtil` to `CommonTestUtils.waitUntilCondition`. 
3. If 2, we could also provide `Duration` overloads for all exiting `waitUntilCondition` overloads.

What do you think ? The last two options will make this a bit fragment, but all related to `waitXyz`.

Reply:
Let's just leave it as it is for now. It is correct, and the slight improvement with Deadline is not worth the big dependency changes.